components/Stands/StandCard.tsx

```diff
@@ -0,0 +1,305 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import * as _ from 'lodash';
+import { NudgeDiv } from '../Styled';
+import { useRouter } from 'next/router';
+import dayjs from 'dayjs';
+
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+import { timeIsBetween, timeIsAfter } from '../../utils/time';
+import { createFragmentContainer, graphql } from 'react-relay';
+import { StandCard_company } from '../../__generated__/StandCard_company.graphql';
+
+dayjs.extend(customParseFormat);
+
+type Package = 'standard' | 'sp' | 'hsp';
+interface StandCardProps {
+  type: Package;
+  time: number;
+  company: StandCard_company;
+}
+
+interface LiveProps {
+  active: boolean;
+}
+
+const getCurrentEvent = (events: any, time: number) => {
+  const currentEvent = events.find((event: any) =>
+    timeIsBetween(time, event.timeStart, event.timeEnd, event.date)
+  );
+  return currentEvent ?? null;
+};
+
+const eventTime = (event: any, truncLength = 50) => {
+  const timeRange = event
+    ? `${event.timeStart.slice(0, 5)} - ${event.timeEnd.slice(0, 5)}`
+    : '';
+  const eventTitle = _.truncate(event ? event.title : '💁🏼‍♀️', {
+    length: truncLength,
+  });
+  return {
+    timeRange,
+    eventTitle,
+  };
+};
+
+const StandCard = ({ company, time, type }: StandCardProps) => {
+  const [currentEvent, setCurrentEvent] = useState();
+  const router = useRouter();
+  const [shouldBreak, setShouldBreak] = React.useState(false);
+
+  useEffect(() => {
+    function onWidthChange(e) {
+      if (!shouldBreak && e.target.innerWidth <= 1199) {
+        setShouldBreak(true);
+      } else if (shouldBreak && e.target.innerWidth > 1199) {
+        setShouldBreak(false);
+      }
+    }
+
+    window.innerWidth > 1199 ? setShouldBreak(false) : setShouldBreak(true);
+
+    window.addEventListener('resize', onWidthChange);
+    return () => {
+      window.removeEventListener('resize', onWidthChange);
+    };
+  });
```

Comment:
Why not do this with css. This seems like very unnecessary updating the component to me, when it could just be completely server rendered and be responsive with css media queries instead.

Reply:
I tried implementing it with css-breakpoint, but it's more than just css-styling that's changed on breakpoint, it's the entire component. Do you have another way to fix this with media queries? @LudvigHz

Reply:
The way I usually do this (and I think the better way to go about it, is by creating two components for the different device targets. I.e. a `<MobileHSPContainer />` and a `<HSPContainer />`, render both and then in the css just set `display: none` and `display: block` on the respective components with media queries.

See an example here: https://github.com/webkom/lego-webapp/blob/9d84f9feb8d79554040355740c61e80ddc8b8fdf/app/routes/overview/components/Overview.js#L202-L227 (JS) and https://github.com/webkom/lego-webapp/blob/9d84f9feb8d79554040355740c61e80ddc8b8fdf/app/routes/overview/components/Overview.css#L126 (CSS). It's necessarily the correct answer, but it mitigates re-rendering and instead puts all the work at the css/browser instead.